app/TEMP_classes_workspace.js

```diff
@@ -66,17 +66,12 @@ class ScaleDegree {
         this.frequency = this.#frequency()
     }
     #frequency() {
-        let frequency = this.inputFrequency;
-        const ratio = this.#ratio();
-        frequency *= ratio; // if no ratio is provided this will be 1
-        frequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
-        return frequency;
+        if (this.modifier.ratioNumerator > 0 && this.modifier.ratioDenominator > 0) return this.#ratio(this.startingFreq);
```

Comment:
re-write so both ratio & detune by cents can be used

Reply:
adding this to the backlog